src/app/guards/verification/email-verified.guard.ts

```diff
@@ -0,0 +1,25 @@
+import { CanActivateFn, Router } from '@angular/router';
+import { AuthGuard } from '../auth.guard';
+import { UserService } from '../../auth/services/user.service';
+import { inject } from '@angular/core';
+import { map } from 'rxjs/operators';
+
+export const EmailVerificationGuard: CanActivateFn = (route, state) => {
+  const userService = inject(UserService);
+  const router = inject(Router);
+
+  const loggedInResult = AuthGuard(route, state);
+  if (loggedInResult !== true) {
+    return loggedInResult;
+  }
+
```

Comment:
- We don't have to repeat this logic as it is implemented in `AuthUnverifiedGuard`